### PR TITLE
JS SDK v2 Announcement post

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

File: _posts/2018-08-15-JS-SDK.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+title: The JS SDK hits version 2.0
+date: 2018-08-15 15:16 -0400
+comments: true
+author: flovilmart
+categories: [announcements, javascript, sdk]
+---
+
+The summer has been really busy and we've dusted out the JS SDK and striped large parts of it's heritage. `Parse.Promises` and `Backbone` style callbacks are definitely out.
```
Comment (acinader) [inline _posts/2018-08-15-JS-SDK.md]: The summer has been really busy and we've dusted off the JS SDK, stripping out large parts of its heritage. `Parse.Promises` and `Backbone` style callbacks are definitely out.

File: _posts/2018-08-15-JS-SDK.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+<!-- more -->
+
+## *Why the move now you may ask?*
+
+Well, we're also working hard on parse-server version 3.0, which revamps cloud code interface. The Cloud Code 'callbacks' have always been in the form: `response.success('yay');` and `response.error('oops');`.
```
Comment (acinader) [inline _posts/2018-08-15-JS-SDK.md]: ...which revamps the cloud...

File: _posts/2018-08-15-JS-SDK.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+The `response` object is following the backbone style callback form, with `success` and `error` callbacks. The next generation of Cloud Code will get rid of this interface that belongs to another age. In 2018, `async` functions and `Promise` are part of the JavaScript language, are available in the LTS node runtime, and not only the leading edge.
```
Comment (acinader) [inline _posts/2018-08-15-JS-SDK.md]: ...JavaScript language, available in the LTS node runtime, and not only the leading edge.

File: _posts/2018-08-15-JS-SDK.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## *What should I do?*
+
+- If you're not using `Parse.Promise`'s with the JavaScript SDK, your code will break. You'll need to rewrite your code from the backbone style success/error callbacks to use promises.
```
Comment (acinader) [inline _posts/2018-08-15-JS-SDK.md]: If you're using...You'll also need to rewrite ...

File: _posts/2018-08-15-JS-SDK.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+- If you're using `Parse.Promise`'s the migration should be minimal. `Parse.Promise` was compatible to a certain extent with the native `Promise` interface.
```
Comment (acinader) [inline _posts/2018-08-15-JS-SDK.md]: If you're using `Parse.Promise`'s the migration should be painless. `Parse.Promise` was mostly compatible with the native `Promise` interface making the transition pretty straightforward.

File: _posts/2018-08-15-JS-SDK.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+In both cases, you'll need to spend some time making sure your code still work appropriately.
```
Comment (acinader) [inline _posts/2018-08-15-JS-SDK.md]: ... your code still works as expected.

File: _posts/2018-08-15-JS-SDK.md (hunk 4 above)
Comment (acinader) [reply _posts/2018-08-15-JS-SDK.md]: I may not understand what you're trying to say here.

File: _posts/2018-08-15-JS-SDK.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+## *It's a lot of work, should I really move to SDK 2.0?*
+
+Yes it is, but it will be a `red diff`, less lines, less code, best code.
```
Comment (acinader) [inline _posts/2018-08-15-JS-SDK.md]: red diff?

File: _posts/2018-08-15-JS-SDK.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  })
+}
+```
+
+With the modern SDK you get the equivalent with just a fraction of the lines:
+
+```js
+async function loginAndGetFeed(username, password,) {
+  await Parse.User.logIn(username, password);
+  return await new Parse.Query('Feed').find();
```
Comment (acinader) [inline _posts/2018-08-15-JS-SDK.md]: eslint doesn't like `return await ...` in this case you don't need the return.

File: _posts/2018-08-15-JS-SDK.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+}
+```
+
+## *I'm hooked Where do I get started?*
```
Comment (acinader) [inline _posts/2018-08-15-JS-SDK.md]: I'm sold.  How do I get started?

**Review by acinader — CHANGES_REQUESTED** [review]
some word smithery

**Comment by flovilmart** [comment]
Thanks @acinader, you’re a damn good editor. I wish I had this style of writing on the 1st shot.

File: _posts/2018-08-15-JS-SDK.md (hunk 8 above)
Comment (flovilmart) [reply _posts/2018-08-15-JS-SDK.md]: ah right you don't need `await`

**Comment by flovilmart** [comment]
@acinader I've made the nits, let me know if I missed anything.

File: _posts/2018-08-15-JS-SDK.md [hunk 10]
```diff
@@ -19,15 +19,15 @@ The `response` object is following the backbone style callback form, with `succe
...
-- If you're using `Parse.Promise`'s the migration should be painless. `Parse.Promise` was mostly compatible with the native `Promise` interface.
+- If you're using `Parse.Promise` the migration to 2.0 should be quick and painless as `Parse.Promise` was mostly compatible with the native `Promise` interface.
 
-In both cases, you'll need to spend some time making sure your code still work as expected. If you have unit tests, it's also time to update them!
+In both cases, you'll need to spend some time making sure your code still works as expected. If you have unit tests, it's also time to update them!
 
 ## *It's a lot of work, should I really move to SDK 2.0?*
 
-Yes it is! It will remove many lines of code, and make your code more readable.
+Yes you should! You'll remove many lines, and making your code more readable.
```
Comment (flovilmart) [inline _posts/2018-08-15-JS-SDK.md]: and making? Should be 'remove' and 'make' no?

File: _posts/2018-08-15-JS-SDK.md [hunk 11]
```diff
@@ -57,15 +57,14 @@ With the modern SDK you get the equivalent with just a fraction of the lines:
 ```js
 async function loginAndGetFeed(username, password,) {
   await Parse.User.logIn(username, password);
-  const feed = await new Parse.Query('Feed').find();
-  return feed;
+  return new Parse.Query('Feed').find();
```
Comment (flovilmart) [inline _posts/2018-08-15-JS-SDK.md]: wanted to make it clear that you get the feed out of it, and then return it. 

Here it looks like you're returning a 'simple' promise.

Comment (acinader) [reply _posts/2018-08-15-JS-SDK.md]: Well, in either case, its the same.  Also, we're promoting fewer lines as a benefit, this is fewer.

Having said that, I trust your judgment, so go with what you think is best.

**Review by acinader — APPROVED** [review]

File: _posts/2018-08-15-JS-SDK.md (hunk 11 above)
Comment (flovilmart) [reply _posts/2018-08-15-JS-SDK.md]: I know it's exactly the same. I don't have the energy to change it back so let's go with this solution.

Just wanted to point out that I wrote it that way on purpose, to show that you get a list of ParseObject as a result when one `await Query.find()`.

Many users are still using the backbone API, most havent heard of async / await. :)

**Pull request merged**